services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py

```diff
@@ -137,6 +138,37 @@ def create(cls, app: FastAPI):
         return table
 
 
+_NEW_ENVIRONMENTS: Final = {
+    "OSPARC_API_BASE_URL": "$OSPARC_VARIABLE_API_HOST",
+    "OSPARC_API_KEY": "$OSPARC_VARIABLE_API_KEY",
+    "OSPARC_API_SECRET": "$OSPARC_VARIABLE_API_SECRET",
+    "OSPARC_STUDY_ID": "$OSPARC_VARIABLE_STUDY_UUID",
+    "OSPARC_NODE_ID": "$OSPARC_VARIABLE_NODE_ID",
+}
+
+
+def auto_inject_environments(
+    compose_spec: ComposeSpecLabelDict,
+) -> ComposeSpecLabelDict:
+    # SEE https://github.com/ITISFoundation/osparc-simcore/issues/5925
+    for service in compose_spec.get("services", {}).values():
```

Comment:
I think it's better to inject it to all the containers. There isn't anything special between applications that run in 1 container or multiple containers.